### PR TITLE
Check return from report.Event and migrate to use ReportingMetricSetV2Error

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
Instead of ignoring the return from report.Error, we should actually check the return to see if report event succeed or not.  Please see https://github.com/elastic/beats/issues/11666 for more details on this. Thanks @fearful-symmetry to bring this up.

This PR also removes `moduleConfig.Period` check in s3 metricset code because it's already done in
`NewMetricSet` in aws.go.

### CODE REVIEW

File: x-pack/metricbeat/module/aws/ec2/_meta/data.json [hunk 1]
```diff
@@ -51,12 +47,12 @@
...
-                    "bytes": 56,
-                    "packets": 1
+                    "bytes": 615.2,
+                    "packets": 6.2
                 },
                 "out": {
-                    "bytes": 88,
-                    "packets": 1.6
+                    "bytes": 841.4,
+                    "packets": 6.8
```
Comment (exekias) [inline x-pack/metricbeat/module/aws/ec2/_meta/data.json]: is this update intended? doesn't seem related to the PR description

**Comment by fearful-symmetry** [comment]
This looks good, as the added logging/reporting steps only happen inside a loop with `continue`. I'm not sure if we need to log something when the metricset closes, but @ruflin may disagree

File: x-pack/metricbeat/module/aws/ec2/_meta/data.json (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/ec2/_meta/data.json]: @exekias Yes I was just trying to run generate data.json to verify the change here still works 😄

File: x-pack/metricbeat/module/aws/ec2/ec2.go [hunk 2]
```diff
@@ -129,9 +127,14 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 				report.Error(err)
 				continue
 			}
-			report.Event(event)
+
+			if reported := report.Event(event); !reported {
+				return errors.Wrap(err, "Error trying to emit event")
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/ec2/ec2.go]: Thanks @jsoriano for the explanation!

File: x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go [hunk 3]
```diff
@@ -38,25 +37,18 @@ func init() {
...
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Beta("The aws s3_daily_storage metricset is beta.")
-	s3Logger := logp.NewLogger(aws.ModuleName)
 
 	moduleConfig := aws.Config{}
 	if err := base.Module().UnpackConfig(&moduleConfig); err != nil {
 		return nil, err
 	}
 
-	if moduleConfig.Period == "" {
```
Comment (ruflin) [inline x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go]: How is this change related?

File: x-pack/metricbeat/module/aws/s3_request/s3_request.go [hunk 4]
```diff
@@ -37,25 +36,18 @@ func init() {
...
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Beta("The aws s3_request metricset is beta.")
-	s3Logger := logp.NewLogger(aws.ModuleName)
 
 	moduleConfig := aws.Config{}
 	if err := base.Module().UnpackConfig(&moduleConfig); err != nil {
 		return nil, err
 	}
 
-	if moduleConfig.Period == "" {
```
Comment (ruflin) [inline x-pack/metricbeat/module/aws/s3_request/s3_request.go]: See above.

File: x-pack/metricbeat/module/aws/sqs/sqs.go [hunk 5]
```diff
@@ -9,6 +9,8 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
```
Comment (ruflin) [inline x-pack/metricbeat/module/aws/sqs/sqs.go]: Nit: move to line ~21

File: x-pack/metricbeat/module/aws/sqs/sqs.go [hunk 6]
```diff
@@ -116,14 +114,20 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
...
 			report.Error(err)
 			continue
 		}
 
 		// Create Cloudwatch Events for SQS
-		createSQSEvents(queueURLs, metricDataResults, regionName, report)
+		err = createSQSEvents(queueURLs, metricDataResults, regionName, report)
+		if err != nil {
+			m.Logger().Debug("Error trying to emit event")
+			return nil
```
Comment (ruflin) [inline x-pack/metricbeat/module/aws/sqs/sqs.go]: Shouldn't it be `continue` here so it continues to the other regions. Or is an error returned if publish event is false?

File: x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go (hunk 3 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go]: Not related. Sorry forgot to update on the PR . This check is not needed here because it's already done in `aws.NewMetricSet(base)` in line 60.

File: x-pack/metricbeat/module/aws/sqs/sqs.go (hunk 6 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/sqs/sqs.go]: The only time the error here is not nil is when report.Event failed, which means the module is being stopped. So instead of `continue` to other regions to collect data which will not be used, `return` seems to be appropriate here.

File: x-pack/metricbeat/module/aws/sqs/sqs.go [hunk 7]
```diff
@@ -219,6 +222,11 @@ func createSQSEvents(queueURLs []string, metricDataResults []cloudwatch.MetricDa
 			report.Event(event)
 			continue
 		}
-		report.Event(event)
+
+		if reported := report.Event(event); !reported {
+			return errors.Wrap(err, "Error trying to emit event")
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/sqs/sqs.go]: yeah done! thanks!

File: x-pack/metricbeat/module/aws/sqs/sqs.go (hunk 5 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/sqs/sqs.go]: done

File: x-pack/metricbeat/module/aws/s3_request/s3_request.go [hunk 8]
```diff
@@ -124,14 +112,20 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
...
-				m.logger.Error(err.Error())
+				m.Logger().Error(err.Error())
 				event.Error = err
 				report.Event(event)
 				continue
 			}
-			report.Event(event)
+
+			if reported := report.Event(event); !reported {
+				m.Logger().Debug("Error trying to emit event")
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/s3_request/s3_request.go]: Nit. These are not errors, the message should be something like "fetch interrupted, metricset stop" or similar.

Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/s3_request/s3_request.go]: @jsoriano  Good point, I changed it to "Fetch interrupted, failed to emit event". Thank you!
